utils/clflags.mli

```diff
@@ -235,11 +235,11 @@ val insn_sched : bool ref
 val insn_sched_default : bool
 
 module Compiler_pass : sig
-  type t = Parsing | Typing
+  type t = Parsing | Typing | Scheduling
   val of_string : string -> t option
   val to_string : t -> string
-  val passes : t list
-  val pass_names : string list
+  val is_compilation_pass : t -> bool
+  val stop_after_pass_names : native:bool -> string list
```

Comment:
I think that @Octachron's suggestion for `available_pass_names` is nicer than the currently proposed API (it's not clear here what in this function is specific to `stop_after`; I remember @gretay-js mentioning that for some passes we want to say "before" rather than "after", but why would that specific function be affected by this difference?)

Reply:
> I think that @Octachron's suggestion for available_pass_names is nicer than the currently proposed API (it's not clear here what in this function is specific to stop_after;

Ok, thanks, I'll change that for this PR and (adjust in the follow up PRs if needed).

> I remember @gretay-js mentioning that for some passes we want to say "before" rather than "after", but why would that specific function be affected by this difference?)

That was about calling the command-line option "-stop-before emit" rather than
"-stop-after scheduling" as scheduling is irrelevant other than it happens to be the one before emit, but for this advanced option, I supposed it should be fine to mention scheduling, to fit with existing interface.